drivers/gpio/gpio_nrfx.c

```diff
@@ -163,6 +166,10 @@ static int gpio_nrfx_pin_configure(const struct device *port, gpio_pin_t pin,
 		nrf_gpio_port_out_clear(cfg->port, BIT(pin));
 	}
 
+#ifdef CONFIG_GPIO_GET_CONFIG
+	get_port_data(port)->output_init = (flags & GPIO_OUTPUT_INIT_HIGH) ? 1 : 0;
```

Comment:
Initialization values are specified per pin, not per port, so they would need to be stored separately for all pins. But wouldn't it be easier to just return the current output value in `gpio_nrfx_pin_get_config()`? I took a quick look at some other drivers and they do so. The 2pin test would pass then (and actually, the test itself seems to be not perfect in this regard, as `GPIO_OUTPUT_INIT_*` flags are not really part of a pin configuration, so they should be filtered out from the value returned from `gpio_pin_get_config()`).